packages/esm-service-queues-app/src/hooks/useQueueEntries.ts

```diff
@@ -99,6 +99,11 @@ export function useQueueEntries(searchCriteria?: QueueEntrySearchCriteria, rep:
       if (pageData?.data?.totalCount && pageData?.data?.totalCount !== totalCount) {
         setTotalCount(pageData?.data?.totalCount);
       }
+      //edge case when the new Data is an empty array E.g. you deleted the last patient, or you cleared the queue
+      if (pageData?.data?.totalCount === 0) {
+        setTotalCount(0);
+        setData([]);
+      }
       if (pageData?.data?.results?.length) {
```

Comment:
```suggestion
      if (pageData?.data?.totalCount > -1 && pageData?.data?.totalCount !== totalCount) {
        setTotalCount(pageData?.data?.totalCount);
      }
      if (pageData?.data?.results) {
```

Try this and let me know how it goes

Reply:
@brandones this worked! 
Thanks!